entrypoint.sh

```diff
@@ -1,13 +1,33 @@
 #!/bin/bash -e
 
+_term() {
+	# FILL UP PROCESS SEARCH PATTERN HERE TO FIND PROPER PROCESS FOR SIGINT:
+	pattern="component_container_mt"
+
+	pid_value="$(ps -ax | grep $pattern | grep -v grep | awk '{ print $1 }')"
+	if [ "$pid_value" != "" ]; then
+		pid=$pid_value
+		echo "Send SIGINT to pid $pid"
+	else
+		pid=1
+		echo "Pattern not found, send SIGINT to pid $pid"
+	fi
+	kill -s SIGINT $pid
+}
+trap _term SIGTERM
+
 ROS_FLAGS=""
 if [[ ${SIMULATION+x} != "" ]]; then
-    ROS_FLAGS="use_sim_time:=true ${ROS_FLAGS}"
+	ROS_FLAGS="use_sim_time:=true ${ROS_FLAGS}"
 fi
 
-ros-with-env ros2 launch octomap_server2 octomap_server.py ${ROS_FLAGS}
+ros-with-env ros2 launch octomap_server2 octomap_server.py ${ROS_FLAGS} &
```

Comment:
Now as we start it in the background the exit code will always be success? Few lines later it checks the exit code. But maybe it is ok as this will not be a permanent solution?

Reply:
Proper exit code of child process should be returned by "wait" command.
Process is started as background, then pid is stored to "child". Then scripts calls "wait $child" which return the child process return code when it exits => stored to RESULT variable. That happens at least according to documentation, however, I haven't verified it fully...